### PR TITLE
Give a try to github actions to build the gem.

### PR DESCRIPTION
**Pull request by mberlanda**
We may use github actions to run tests and build the gem.
I've never used it before and I was curious to give a try

### CODE REVIEW

**Comment by mberlanda** [comment]
I am just changing the files from the ui. I am going to squash the commits at the end 😄

**Comment by mberlanda** [comment]
It's quite strange that the rake task `build` is not found here:
https://bundler.io/v1.12/guides/creating_gem.html#releasing-the-gem

**Comment by mhenrixon** [comment]
No its not strange at all. I removed those tasks because they were causing problems with `gem release` or something.

I couldn't make it work for sidekiq unique jobs so figured why keep it around :)

**Comment by mberlanda** [comment]
What use case would you expect for action then?
We may want to run here linting and tests instead of travis

**Comment by mhenrixon** [comment]
Nah, I'd love to move everything over. Just use the same commands as Travis? 

Btw, have you seen jobs for Travis? We are on https://travis-ci.com, which offers cool stuff (that from my understanding Github Actions does too), which is in regards to handling, for instance, complex workflows like conditionally releasing a gem.

I'm sorry, I didn't intend for my comment to be discouraging. What I was trying to say is that I'd like all that functionality in the release process. I believe we are supposed to use either Github actions or Travis. 

We can also release the gem from Travis, no problem. I meant to encourage you to replace Travis with Github. 

I'd love to have everything on GitHub. 

- Packages (gem releases)
- Actions (CI/CD)
- Source Code
- Documentation

Why spread it out if we can avoid it?

**Comment by mberlanda** [comment]
The only downside of putting packages on gihub is that the consumer will need to add another gem source in the Gemfile and as per today even public packages require authentication (I contacted the support last week for a docker image published publicly unintentionally 😄 )

**Comment by mhenrixon** [comment]
> The only downside of putting packages on gihub is that the consumer will need to add another gem source in the Gemfile and as per today even public packages require authentication

@mberlanda then I suggest we keep everything as is but move the build+test stages to Actions. I'm just a little hesitant about the release process because it isn't air-tight yet. I did my best with the scripts but occasionally I run into some weird issue because some file wasn't staged.

Happy to continue that work though, I mean if you get it right in one project you could create a gem out of it that all projects could use?

Another gem for the organization? 💃

File: .github/workflows/gempush.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    branches:
+      - master
+
+jobs:
+  build:
+    name: Build + Publish
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@master
```
Comment (mhenrixon) [inline .github/workflows/gempush.yml]: I do not understand why it is done this way by all generators... I prefer to indend the `-` two spaces to the rig Iht. It reads much better to me that way. Like you have above on `on`:

```yaml
on:
  pull_request:
    branches:
      - master
  push:
    branches:
      - master
```

File: .github/workflows/gempush.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        # mkdir -p $HOME/.gem
+        # touch $HOME/.gem/credentials
+        # chmod 0600 $HOME/.gem/credentials
+        # printf -- "---\n:github: Bearer ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
+        # gem build *.gemspec
+        # gem push --KEY github --host https://rubygems.pkg.github.com/${OWNER} *.gem
+      # env:
+        # GEM_HOST_API_KEY: ${{secrets.GPR_AUTH_TOKEN}}
+        # OWNER: username
+      run: |
```
Comment (mhenrixon) [inline .github/workflows/gempush.yml]: This is one of the reasons I love Actions and want to switch there!! The `run: |` that allows multi lines commands without resorting to weirdness... ❤️

**Review by mhenrixon — COMMENTED** [review]
It looks really nice, to be honest cleaner than Travis.

What I am missing is: 

- Slack notifications
- Email notifications to my personal email
- Testing styles with reek and rubocop (preferably just `rake default` or something equally simple)
- Code Climate Test coverage reporting

**Comment by mberlanda** [comment]
@mhenrixon After a more attentive investigation I have some more points about what to use for CI/CD

Github Actions vs Travis
`+` Clear multi step workflow
`+` Stateful multistep (avoid a lot of repetition)
`-` Complex to adapt to test matrix (multiple gemfiles, ruby versions, redis versions etc.)

We may stick on Travis for now and consider this solution later. (Even if in terms of execution time it is really fast)

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+          gem install bundler
+          bundle install --quiet
+      - name: Codeclimate Before Build
+        run: |
+          curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+          chmod +x ./cc-test-reporter
+          ./cc-test-reporter before-build
+      - name: RSpec tests
+        run: |
+          bundle exec rspec --require spec_helper --tag ~perf
```
Comment (mhenrixon) [inline .github/workflows/tests.yml:37]: Maybe run `bundle exec appraisal rspec`? That would run all gemfiles

**Comment by mhenrixon** [comment]
Thank you for the investigation @mberlanda. I think we could totally pull it off with using `bundle exec rake appraisal rspec` but it does make any failures kind of difficult to discover.

Let's stick to Travis for now then.

**Comment by mberlanda** [comment]
We can close for now then. We will eventually reconsider it later

**Pull request closed without merging**